rfd/XXXX-securing-amis.md

```diff
@@ -0,0 +1,225 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: Draft
+---
+
+# RFD 0XXX - Delivering more secure AMIs
+
+# Required Approvers
+* Engineering: @r0mant
+* Security: (@reedloden && @jentfoo)
+* Product: (@xinding33 || @klizhentas)
+
+## What
+
+This RFD discusses structures and processes to increase the security of the
+Amazon EC2 Disk Images (a.k.a AMIs) used to ship Teleport to customers.
+
+## Why
+
+One way our customers consume Teleport is by using a pre-built and  -configured
+AMI to run Teleport in AWS EC2 instances. As a provider of these images we are 
+(at least partially) responsible for the secutity of all ssoftware included in
+the image, not just teleport. 
+
+We should endeavour to not ship vulnerabilities to our clients, even if those
+vulnerabilities are not directly in our software.
+
+This is not simply an academic exercise. Customers have been asking for hardened
+AMIs - especially sor AMIs that conform to well-known security beckhmarks. See 
+[here](https://github.com/gravitational/teleport/issues/8078) for an example.
+
+## Details
+
+### Current Situation
+
+We  currently use [Packer](https://www.hashicorp.com/products/packer) and a shell
+script to provision our AMIs, based on Amazon Linux 2. The same Packerfile is used
+as a basis for several AMIs
+
+Inclusions for all AMIs:
+  * [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/)
+    for capturing monitoring data
+  * [InfluxDB](https://www.influxdata.com/) as a data store for Telegraf
+  * [Grafana](https://grafana.com/) to display time series data
+  * [nginx](https://nginx.com) to serve Grafana's front end  
+  * EFF [`certbot`](https://https://certbot.eff.org/) for ACME Certificate
+    Rotation (used by optional ssm `systemd` units)
+  * Python 3 and `pip` for installing and running `certbot`
+  * `uuid` for token generation (used by optional ssm `systemd` units)
+  * `gcc` for ???
```

Comment:
In general I don't believe we should be providing compilers or build tooling on these AMI's

Reply:
Once I find out what they're for, I can be a bit more decisive about getting rid of them. Right now I _assume_ that to  required to natively compile some node module in the monitoring stack, but that's a wild guess with no science behind it.

Reply:
I can't actually remember why they're there either. `uuid` is used to generate random join tokens as has been noted elsewhere, but I presume `gcc` was a dependency that was pulled in for something else.